vertical-pod-autoscaler/pkg/utils/vpa/limit_and_request_scaling.go

```diff
@@ -103,21 +103,37 @@ func GetBoundaryRequest(originalRequest, originalLimit, boundaryLimit, defaultLi
 	if originalRequest == nil || originalRequest.Value() == 0 {
 		return boundaryLimit
 	}
-	result, _ := scaleQuantityProportionally(originalRequest /* scaledQuantity */, originalLimit /*scaleBase*/, boundaryLimit /*scaleResult*/)
+	result, _ := scaleQuantityProportionally(originalRequest /* scaledQuantity */, originalLimit /*scaleBase*/, boundaryLimit /*scaleResult*/, noRounding)
 	return result
 }
 
+type roundingMode int
+
+const (
+	noRounding roundingMode = iota
+	roundUpToFullUnit
+	roundDownToFullUnit
+)
+
 // scaleQuantityProportionally returns value which has the same proportion to scaledQuantity as scaleResult has to scaleBase
 // It also returns a bool indicating if it had to cap result to MaxInt64 milliunits.
-func scaleQuantityProportionally(scaledQuantity, scaleBase, scaleResult *resource.Quantity) (*resource.Quantity, bool) {
+func scaleQuantityProportionally(scaledQuantity, scaleBase, scaleResult *resource.Quantity, rounding roundingMode) (*resource.Quantity, bool) {
 	originalMilli := big.NewInt(scaledQuantity.MilliValue())
```

Comment:
If we worked on Value() for memory instead of MilliValue, would we still need rounding?

Reply:
We wouldn't have to worry about rounding down. But we'd have to worry about rounding up.